File: backend/src/main/java/com/guia/bandeiras/service/NotificationService.java
package com.guia.bandeiras.service;

import com.guia.bandeiras.dto.Notification;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Slf4j
@Service
public class NotificationService {
    
    private final List<Notification> notifications = new ArrayList<>();
    
    public NotificationService() {
        // Initialize with some sample notifications
        notifications.add(new Notification(
            1L,
            "Nova atualização no manual de Chargeback da Visa disponível",
            "info",
            false,
            LocalDateTime.now().minusDays(1)
        ));
        notifications.add(new Notification(
            2L,
            "Manual de MCC do MasterCard atualizado com novas categorias",
            "warning",
            false,
            LocalDateTime.now().minusDays(3)
        ));
        notifications.add(new Notification(
            3L,
            "Novas regras de 3D Secure implementadas",
            "success",
            true,
            LocalDateTime.now().minusDays(5)
        ));
    }
    
    public List<Notification> getAllNotifications() {
        log.info("Fetching all notifications");
        return notifications;
    }
    
    public List<Notification> getUnreadNotifications() {
        log.info("Fetching unread notifications");
        return notifications.stream()
                .filter(n -> !n.getRead())
                .toList();
    }
    
    public void markAsRead(Long id) {
        log.info("Marking notification {} as read", id);
        notifications.stream()
                .filter(n -> n.getId().equals(id))
                .findFirst()
                .ifPresent(n -> n.setRead(true));
    }
    
    public void markAllAsRead() {
        log.info("Marking all notifications as read");
        notifications.forEach(n -> n.setRead(true));
    }
    
    public void addNotification(String message, String type) {
        log.info("Adding notification: {} - {}", type, message);
        Long newId = notifications.stream()
                .mapToLong(Notification::getId)
                .max()
                .orElse(0L) + 1;
        
        notifications.add(new Notification(
            newId,
            message,
            type,
            false,
            LocalDateTime.now()
        ));
    }
}

File: frontend/src/app/services/notification.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';

export interface Notification {
  id: number;
  message: string;
  type: 'info' | 'success' | 'warning' | 'error';
  timestamp: Date;
  read: boolean;
}

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private notifications = new BehaviorSubject<Notification[]>([]);
  notifications$ = this.notifications.asObservable();
  private unreadCount = new BehaviorSubject<number>(0);
  unreadCount$ = this.unreadCount.asObservable();

  constructor() {
    this.loadNotifications();
    this.simulateManualUpdates();
  }

  private loadNotifications(): void {
    const saved = localStorage.getItem('notifications');
    if (saved) {
      this.notifications.next(JSON.parse(saved));
      this.updateUnreadCount();
    }
  }

  private saveNotifications(): void {
    localStorage.setItem('notifications', JSON.stringify(this.notifications.value));
  }

  private updateUnreadCount(): void {
    const unread = this.notifications.value.filter(n => !n.read).length;
    this.unreadCount.next(unread);
  }

  addNotification(message: string, type: 'info' | 'success' | 'warning' | 'error' = 'info'): void {
    const notification: Notification = {
      id: Date.now(),
      message,
      type,
      timestamp: new Date(),
      read: false
    };
    
    const current = this.notifications.value;
    this.notifications.next([notification, ...current].slice(0, 50));
    this.saveNotifications();
    this.updateUnreadCount();
  }

  markAsRead(id: number): void {
    const current = this.notifications.value.map(n => 
      n.id === id ? { ...n, read: true } : n
    );
    this.notifications.next(current);
    this.saveNotifications();
    this.updateUnreadCount();
  }

  markAllAsRead(): void {
    const current = this.notifications.value.map(n => ({ ...n, read: true }));
    this.notifications.next(current);
    this.saveNotifications();
    this.updateUnreadCount();
  }

  clearAll(): void {
    this.notifications.next([]);
    this.saveNotifications();
    this.updateUnreadCount();
  }

  private simulateManualUpdates(): void {
    setTimeout(() => {
      this.addNotification('Novo manual da Visa disponível: Atualização de regras de chargeback', 'info');
    }, 5000);

    setTimeout(() => {
      this.addNotification('MasterCard: Novas diretrizes de segurança implementadas', 'warning');
    }, 15000);
  }
}
